ckan/tests/logic/action/test_create.py

```diff
@@ -443,6 +443,64 @@ def test_doesnt_require_url(self):
 
         assert not stored_resource['url']
 
+    def test_mimetype_by_url(self):
+        context = {}
+        params = {
+            'package_id': factories.Dataset()['id'],
+            'url': 'http://localhost/data.csv',
+            'name': 'A nice resource',
+        }
+        result = helpers.call_action('resource_create', context, **params)
+
+        mimetype = result.pop('mimetype')
+        assert_equals(mimetype, 'text/csv')
+        #maybe see if the mimetype is in the available list of mimetypes?
+
+    def test_mimetype_by_user(self):
+        context = {}
+        params = {
+            'package_id': factories.Dataset()['id'],
+            'url': 'http://localhost/data.csv',
+            'name': 'A nice resource',
+            'mimetype': 'application/csv'
+        }
+        result = helpers.call_action('resource_create', context, **params)
+
+        mimetype = result.pop('mimetype')
+        assert_equals(mimetype, 'application/csv')
+
+    def test_mimetype_by_upload(self):
+        pass
+
+    def test_size_of_resource_by_upload(self):
+        pass
+        '''
+        context = {}
+        params = {
+            'package_id': factories.Dataset()['id'],
+            'url': 'http://data',
+            'name': 'A nice resource',
+            'upload': file('/home/vagrant/test.txt', 'wb+')
```

Comment:
@TkTech I'm trying with a fake FieldStorage class, but somewhere in the test it's failing and the error message is a red herring and not telling me the actual fail point?

```
FAIL: ckan.tests.logic.action.test_create.TestResourceCreate.test_size_of_resource_by_upload
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/ckan/default/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/vagrant/ckan/lib/default/src/ckan/ckan/tests/logic/action/test_create.py", line 498, in test_size_of_resource_by_upload
    assert size
AssertionError:
-------------------- >> begin captured stdout << ---------------------
FieldStorage('upload', 'test.txt')
{'name': 'A nice resource',
 'package_id': u'fdb3bb89-bd74-46ae-bdbb-44b896f7c7e8',
 'upload': FieldStorage('upload', 'test.txt'),
 'url': 'http://data'}

--------------------- >> end captured stdout << ----------------------
```

```
def test_size_of_resource_by_upload(self):
        import cgi
        class FakeFileStorage(cgi.FieldStorage):
            def __init__(self, fp, filename):
                self.file = fp
                self.filename = filename
                self.name = 'upload'


        with open('/home/vagrant/test.txt', 'rb') as test_file:
            test_resource = FakeFileStorage(test_file, 'test.txt')

        print(test_resource)
        context = {}
        params = {
            'package_id': factories.Dataset()['id'],
            'url': 'http://data',
            'name': 'A nice resource',
            'upload': test_resource
        }
        result = helpers.call_action('resource_create', context, **params)

        size = result.pop('size')
        assert size
```